src/empirical.cpp

```diff
@@ -5,9 +5,9 @@ using std::string;
 // Obtain the empirical given beta, beta_vcov, theta.
 List get_empirical(List mmrm_data, NumericVector theta, NumericVector beta, NumericMatrix beta_vcov, string type) {
   NumericMatrix x = mmrm_data["x_matrix"];
-  matrix<double> x_matrix = as_matrix(x);
+  matrix<double> x_matrix = as_matrix<matrix<double>, NumericMatrix>(x);
```

Comment:
this repeats quite a bit below, similarly for vector, maybe it helps to define a quick helper like `as_num_matrix()`?

Reply:
it is two way conversion between matrix<double> and  NumericMatrix , and we will need two quick helpers for this if needed

Reply:
another possibility is to use `as_matrix<matrix<double>>(x)` here, and the second typename can be deduced from the type of x.

Reply:
or, if this is still too long, we can use some alias to make it shorter. Anyway we use templates to avoid duplicated code so it is still not so nice to again implement some new functions?

Reply:
yeah maybe an alias. I would just like to make it more easily readable